src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -257,23 +257,35 @@ public Task AnnounceBlocks(List<uint256> blockHashesToAnnounce)
                     {
                         headers.Add(chainedBlock.Header);
                     }
-                    else if (chainBehavior.PendingTip.GetAncestor(chainedBlock.Height) != null)
-                    {
-                        continue;
-                    }
-                    else if (chainBehavior.PendingTip.GetAncestor(chainedBlock.Previous.Height) != null)
-                    {
-                        // Peer doesn't have this header but they do have the prior one.
-                        // Start sending headers.
-                        foundStartingHeader = true;
-                        headers.Add(chainedBlock.Header);
-                    }
                     else
                     {
-                        // Peer doesn't have this header or the prior one -- nothing will
-                        // connect, so bail out.
-                        revertToInv = true;
-                        break;
+                        ChainedBlock peersBlock = chainBehavior.PendingTip.GetAncestor(chainedBlock.Height);
+
+                        if (peersBlock != null)
+                        {
+                            if (peersBlock.HashBlock == chainedBlock.HashBlock)
+                                continue;
+
+                            headers.Add(chainedBlock.Header);
```

Comment:
are you missing `foundStartingHeader = true` here or not?

Reply:
if you do, consider DRY and merge this case with the last case below

Reply:
>are you missing foundStartingHeader = true here or not?

No. This flag is for the cases when some of the hashes are good and some are bad (bad = we think peer won't validate them) so we want to start sending starting from the first one that will be approved.

Reply:
hmm my understanding of this flag was totally different, we should get in consensus on what this is first